File: zowe-rest-api-sample-spring/src/main/java/org/zowe/sample/apiservice/config/ApplicationConfig.java
/*
 * This program and the accompanying materials are made available under the terms of the
 * Apache License, Version 2.0 which accompanies this distribution, and is available at
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */
package org.zowe.sample.apiservice.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.zowe.commons.error.ErrorService;
import org.zowe.commons.error.ErrorServiceImpl;
import org.zowe.commons.spring.ServiceStartupEventHandler;

@Configuration
@ComponentScan("org.zowe.commons")
public class ApplicationConfig implements ApplicationListener<ApplicationReadyEvent> {

    @Value("${apiml.service.title}")
    private String serviceTitle;

    @Autowired
    private ServiceStartupEventHandler serviceStartupEventHandler;

    private final ErrorService errorService = ErrorServiceImpl.getDefault();

    @Bean
    public ErrorService errorService() {
        return errorService;
    }

    @Override
    public void onApplicationEvent(final ApplicationReadyEvent event) {
        serviceStartupEventHandler.onServiceStartup(serviceTitle, ServiceStartupEventHandler.DEFAULT_DELAY_FACTOR);
    }
}

File: zowe-rest-api-sample-kotlin-spring/src/main/kotlin/org/zowe/sample/kotlin/apiservice/config/ApplicationConfig.kt
/*
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */
package org.zowe.sample.kotlin.apiservice.config

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.FilterType
import org.zowe.apiml.enable.EnableApiDiscovery

@Configuration
@EnableApiDiscovery
@ComponentScan(
    basePackages = ["org.zowe.commons.spring"],
    useDefaultFilters = false,
    includeFilters = [ComponentScan.Filter(type = FilterType.REGEX,
        pattern = [
            "org.zowe.commons.spring.CustomRestExceptionHandler",
            "org.zowe.commons.spring.WebConfig"
        ]
    )]
)
class ApplicationConfig {
}